### PR TITLE
tx host env panics fixes

### PR DESCRIPTION
**Pull request by tzemanovic**
## Indicate on which release or other PRs this topic is based on

0.30.1

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

File: crates/namada/src/ledger/vp_host_fns.rs [hunk 1]
```diff
@@ -41,6 +41,8 @@ pub enum RuntimeError {
     ReadPermanentValueError,
     #[error("Invalid transaction code hash")]
     InvalidCodeHash,
+    #[error("No value found in result buffer")]
+    NoValInResultBuffer,
```
Comment (sug0) [inline crates/namada/src/ledger/vp_host_fns.rs:45]:
```suggestion
    NoValueInResultBuffer,
```
nit: IMO there's no reason to contract the identifier here, especially when `l` and `I` look pretty similar

File: crates/namada/src/vm/host_env.rs [hunk 2]
```diff
@@ -72,9 +72,12 @@ pub enum TxRuntimeError {
     MissingTxData,
     #[error("IBC: {0}")]
     Ibc(#[from] namada_ibc::Error),
+    #[error("No value found in result buffer")]
+    NoValInResultBuffer,
```
Comment (sug0) [inline crates/namada/src/vm/host_env.rs:76]:
```suggestion
    NoValueInResultBuffer,
```

File: crates/namada/src/vm/wasm/memory.rs [hunk 3]
```diff
@@ -171,10 +171,10 @@ pub fn write_vp_inputs(
 /// it will try to grow the memory.
 fn check_bounds(memory: &Memory, base_offset: u64, len: usize) -> Result<()> {
     tracing::debug!(
-        "check_bounds pages {}, data_size {}, base_offset + len {}",
+        "check_bounds pages {}, data_size {}, base_offset {base_offset}, len \
+         {len}",
```
Comment (sug0) [inline crates/namada/src/vm/wasm/memory.rs:175]: probably not the most appropriate PR to make this change, but I think renaming `base_offset` to `base_address`, and `len` to `offset` would make more sense. we're calculating an offset over some base address, which we will dereference at some later point in time

File: crates/tests/src/vm_host_env/tx.rs [hunk 4]
```diff
@@ -521,3 +521,293 @@ mod native_tx_host_env {
...
+    fn arb_test_setup() -> impl Strategy<Value = TestSetup> {
+        (
+            any::<bool>(),
+            any::<bool>(),
+            any::<bool>(),
+            namada::types::storage::testing::arb_key(),
+            arb_u64(),
+            arb_u64(),
+            any::<Vec<u8>>(),
+            arb_u64(),
```
Comment (sug0) [inline crates/tests/src/vm_host_env/tx.rs:779]: idk how proptest shrinks results, but half of `u64` values will be invalid. depends on how biased we want the tests to be (over failure or success). maybe it doesn't matter here.

File: crates/tests/src/vm_host_env/tx.rs [hunk 5]
```diff
@@ -521,3 +521,293 @@ mod native_tx_host_env {
...
+
+    fn arb_u64() -> impl Strategy<Value = u64> {
+        prop_oneof![
+            Just(u64::MIN),
+            Just(u64::MIN + 1),
+            Just(u64::MAX),
+            Just(u64::MAX - 1),
+            u64::MIN + 2..u64::MAX - 1,
+        ]
+    }
```
Comment (sug0) [inline crates/tests/src/vm_host_env/tx.rs:812]: oh this is a custom strategy

**Review by sug0 — APPROVED** [review]
some nits and comments. adding proptests to catch panics was a great idea.

File: crates/namada/src/vm/wasm/memory.rs (hunk 3 above)
Comment (tzemanovic) [reply crates/namada/src/vm/wasm/memory.rs]: I'm ok with it :) - done in [b8ca7af](https://github.com/anoma/namada/pull/2401/commits/b8ca7af34d223166be53f161be2ec940cda9a48b)

File: crates/namada/src/ledger/vp_host_fns.rs (hunk 1 above)
Comment (tzemanovic) [reply crates/namada/src/ledger/vp_host_fns.rs]: done in [5dda2da](https://github.com/anoma/namada/pull/2401/commits/5dda2daec6ecab1f55fc95c6d895368d430978c4)

File: crates/tests/src/vm_host_env/tx.rs (hunk 4 above)
Comment (tzemanovic) [reply crates/tests/src/vm_host_env/tx.rs:779]: I've added some weights to the strategy below to change to be more likely to use the valid range. We still do want the invalid values here too though

Comment (tzemanovic) [reply crates/tests/src/vm_host_env/tx.rs:779]: [6e6e0d6](https://github.com/anoma/namada/pull/2401/commits/6e6e0d6523887c0e7030c94c3eb4084029b6bd7d)

**Review by sug0 — APPROVED** [review]

**Pull request merged**